token-lending/program/src/processor.rs

```diff
@@ -750,8 +787,31 @@ fn _redeem_reserve_collateral<'a>(
     }
 
     let liquidity_amount = reserve.redeem_collateral(collateral_amount)?;
+
+    if check_rate_limits {
+        lending_market
+            .rate_limiter
+            .update(
+                clock.slot,
+                reserve.market_value(Decimal::from(liquidity_amount))?,
```

Comment:
It seems like this should be based on worst-case market value right?

Reply:
heh, yeah. i did this PR before adding the 2 prices stuff. will change